subworkflows/local/deepvariant_caller.nf

```diff
@@ -0,0 +1,44 @@
+//
+// Check input samplesheet and get read channels
+//
+
+include { SAMTOOLS_VIEW } from '../../modules/nf-core/samtools/view/main'
+include { DEEPVARIANT }   from '../../modules/nf-core/deepvariant/main'
+
+workflow DEEPVARIANT_CALLER {
+
+    take:
+    reads    // [ val(meta), data, index ]
+    fasta    // path to reference fasta file
+    fai      // path to reference fasta index file 
+    gzi
+    interval // path to interval bed file
+
+    main:
+    ch_versions = Channel.empty()
+
+    SAMTOOLS_VIEW ( reads, fasta, [] )
+    ch_versions = ch_versions.mix ( SAMTOOLS_VIEW.out.versions )
+     
+    cram_crai = SAMTOOLS_VIEW.out.cram
+                .join(SAMTOOLS_VIEW.out.crai)
+
+    cram_crai.map { 
+        if (interval) { it.add ( interval ) } else { it.add( [] ) }
+    }
```

Comment:
Right, you talked about that this morning. I don't like modifying in-place the content of a channel because it may confuse Nextflow and the caching. I would do something like this:

```suggestion
    cram_crai = SAMTOOLS_VIEW.out.cram
                .join(SAMTOOLS_VIEW.out.crai)
                .map { it + [interval ?: []] }
```